subsys/net/ip/route.h

```diff
@@ -54,11 +55,17 @@ struct net_route_entry {
 	/** Network interface for the route. */
 	struct net_if *iface;
 
+	/** Route lifetime timer. */
+	struct net_timeout lifetime;
+
 	/** IPv6 address/prefix of the route. */
 	struct in6_addr addr;
 
 	/** IPv6 address/prefix length. */
 	uint8_t prefix_len;
+
+	/** Is the route valid forever */
+	uint8_t is_infinite : 1;
```

Comment:
Just wondering if we could use the lifetime variable for this, so if lifetime is K_FOREVER, then it is valid forever. What do you think?

Reply:
I thought it would be a reasonable approach as well, it turned out however that the `struct net_timeout` which I've used for the expiry time computation does not support a concept of "infinity".  So I've decided to follow the same pattern as address/prefix/router expiration timer does, and use this extra `is_infinite` variable.

I guess that I could rewrite the lifetime logic to use some different module underneath, like `k_timeout_t`, still, I think it'd then make sense to rework other networking modules as well.

Reply:
No worries, just wondered if this would make sense.